src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -3331,9 +3296,9 @@ def test_HaloReadAccess_input_field():
     with pytest.raises(GenerationError) as excinfo:
         _ = HaloReadAccess(None, None)
     assert (
-        "Generation Error: HaloInfo class expects an argument of type "
-        "DynArgument, or equivalent, on initialisation, but found, "
-        "'%s'" % type(None) in str(excinfo.value))
+        f"Generation Error: HaloInfo class expects an argument of type "
+        f"DynArgument, or equivalent, on initialisation, but found, "
+        f"'{str(type(None))}'" in str(excinfo.value))
```

Comment:
You shouldn't need `str` within an argument to an f-string?

Reply:
Right again, I don't. Removed.